system/lib/wasmfs/wasmfs.cpp

```diff
@@ -63,9 +64,39 @@ extern "C" void wasmfs_flush(void) {
   // Flush musl libc streams.
   fflush(0);
 
-  // Flush our own streams. TODO: flush all backends.
+  // Flush our own streams.
   (void)SpecialFiles::getStdout()->locked().flush();
   (void)SpecialFiles::getStderr()->locked().flush();
+
+  std::stack<std::shared_ptr<Directory>> toFlush;
+  toFlush.push(wasmFS.getRootDirectory());
+  
+  while(!toFlush.empty()) {
```

Comment:
Please run clang-format on that. It should change this for example to `while (` (with a space), and possibly other stuff.